hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSourceManager.java

```diff
@@ -189,13 +189,25 @@ public ReplicationSourceManager(final ReplicationQueues replicationQueues,
    * @param holdLogInZK if true then the log is retained in ZK
    */
   public synchronized void logPositionAndCleanOldLogs(Path log, String id, long position,
-      boolean queueRecovered, boolean holdLogInZK) throws ReplicationSourceWithoutPeerException {
+      boolean queueRecovered, boolean holdLogInZK) {
     String fileName = log.getName();
     this.replicationQueues.setLogPosition(id, fileName, position);
     if (holdLogInZK) {
       return;
     }
-    cleanOldLogs(fileName, id, queueRecovered);
+
+    try {
+      cleanOldLogs(fileName, id, queueRecovered);
+    } catch (ReplicationSourceWithoutPeerException rspe) {
```

Comment:
Should we move handling of `ReplicationSourceWithoutPeerException` exception to `ReplicationSourceManager #cleanOldLogs(SortedSet<String> wals, String key, String id)` ?